UlKit/UlKit/Feed/Controller/FeedViewController.swift

```diff
@@ -0,0 +1,251 @@
+        super.viewDidLoad()
+        setupTableView()
+        addSubview()
+        setupBarButtonItem()
+    }
+
+    override func viewWillLayoutSubviews() {
+        super.viewWillLayoutSubviews()
+        setupConstraints()
+    }
+
+    // MARK: - Private Methods
+
+    private func addSubview() {
+        view.backgroundColor = .white
+        view.addSubview(rmLinkButton)
+        view.addSubview(messagesIconButton)
+        view.addSubview(tableView)
+    }
+
+    private func setupBarButtonItem() {
+        let rmLogo = UIBarButtonItem(customView: rmLinkButton)
+        let messagesIcon = UIBarButtonItem(customView: messagesIconButton)
+
+        rmLogo.customView?.widthAnchor.constraint(equalToConstant: 70).isActive = true
+        rmLogo.customView?.heightAnchor.constraint(equalToConstant: 26).isActive = true
+
+        navigationItem.rightBarButtonItem = messagesIcon
+        navigationItem.leftBarButtonItem = rmLogo
+    }
+
+    private func setupTableView() {
+        tableView.register(FeedStoriesViewCell.self, forCellReuseIdentifier: FeedStoriesViewCell.reuseIdentifier)
+        tableView.register(FeedPostsViewCell.self, forCellReuseIdentifier: FeedPostsViewCell.reuseIdentifier)
+        tableView.register(
+            FeedRecommendationsViewCell.self,
+            forCellReuseIdentifier: FeedRecommendationsViewCell.reuseIdentifier
+        )
+        tableView.rowHeight = UITableView.automaticDimension
+        tableView.estimatedRowHeight = 450
+        tableView.allowsSelection = false
+        tableView.showsVerticalScrollIndicator = false
+        tableView.separatorStyle = .none
+        tableView.translatesAutoresizingMaskIntoConstraints = false
+        tableView.delegate = self
+        tableView.dataSource = self
+        tableView.reloadData()
+    }
+
+    private func setupConstraints() {
+        rmLinkButton.topAnchor.constraint(equalTo: view.topAnchor, constant: 60).isActive = true
+        rmLinkButton.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 15).isActive = true
+
+        messagesIconButton.topAnchor.constraint(equalTo: view.topAnchor, constant: 60).isActive = true
+        messagesIconButton.rightAnchor.constraint(equalTo: view.rightAnchor, constant: -15).isActive = true
+
+        tableView.topAnchor.constraint(equalTo: view.topAnchor, constant: 10).isActive = true
+        tableView.leftAnchor.constraint(equalTo: view.leftAnchor, constant: 0).isActive = true
+        tableView.rightAnchor.constraint(equalTo: view.rightAnchor, constant: 0).isActive = true
+        tableView.bottomAnchor.constraint(equalTo: view.bottomAnchor, constant: 0).isActive = true
+    }
+}
+
+// MARK: - Extensions
+
+extension FeedViewController: UITableViewDataSource {
+    func numberOfSections(in tableView: UITableView) -> Int {
+        rowsType.count
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        let cell = rowsType[section]
+        switch cell {
```

Comment:
```suggestion
        switch rowsType[section] {
```
если так сделать, то дефолт можно будет не прописывать, так поудобней будет при расширении количества секций, свич сразу поругается, что нужно задать количество строк, а не сделает дефектную реализацию

Reply:
Саша попросил сделать default. До этого было без default.

Reply:
Окей